server/src/main/java/org/elasticsearch/cluster/SnapshotDeletionsInProgress.java

```diff
@@ -174,6 +175,8 @@ public Entry(Snapshot snapshot, long startTime, long repositoryStateId) {
             this.snapshot = snapshot;
             this.startTime = startTime;
             this.repositoryStateId = repositoryStateId;
+            assert repositoryStateId > RepositoryData.EMPTY_REPO_GEN :
```

Comment:
We would incorrectly get a `-1` (empty repo gen) here when we deleted/aborted an initializing snapshot (because the repo gen is at `-2` during the snapshot INIT stage). This wouldn't have caused any corruption and only made deletes fail but still, this assertion would've avoided not catching this race for so long :)